graylog2-server/src/main/java/org/graylog2/shared/rest/resources/ProxiedResource.java

```diff
@@ -365,6 +364,45 @@ protected <RemoteInterfaceType, RemoteCallResponseType> NodeResponse<RemoteCallR
         return requestOnLeader(remoteInterfaceFunction, interfaceClass, getDefaultProxyCallTimeout());
     }
 
+    /**
+     * Execute the given remote interface function on the given node.
+     * <p>
+     * This is used to forward an API request to the given node. It is useful in situations where an API call has to
+     * be executed on a particular, chosen node.
+     * <p>
+     * The returned {@link NodeResponse} object is constructed from the remote response's status code and body.
+     */
+    protected <RemoteInterfaceType, RemoteCallResponseType> NodeResponse<RemoteCallResponseType> requestOnNode(
+            String nodeId,
+            Function<RemoteInterfaceType, Call<RemoteCallResponseType>> remoteInterfaceFunction,
+            Class<RemoteInterfaceType> interfaceClass,
+            Duration timeout
+    ) throws IOException {
+
+        final Node executionNode = nodeService.allActive().values().stream()
```

Comment:
How about using the `nodeService#byNodeId(String)` method instead of loading all nodes?

Reply:
It does not guarantee the returned node is active, does it?
Do we have a method to check later on if a single node is active?

Reply:
We do the same for the leader, btw.
I believe it is not a big deal, as there are usually just a few nodes, 10+ being a rare exception?

Reply:
Aye. Let's leave it as it is for now. :+1:

Reply:
Ok!